megamek/src/megamek/common/TankTrailerHitch.java

```diff
@@ -0,0 +1,291 @@
+    protected int getExteriorLocs(boolean isRear) {
+        if (isRear) {
+            return TankTrailerHitch.EXTERIOR_LOCATIONS_REAR;
+        }
+        return TankTrailerHitch.EXTERIOR_LOCATIONS_FRONT;
+    }
+
+    /**
+     * Get the <code>String</code> to report the presence (or lack thereof) of a
+     * towed trailer.
+     * <p/>
+     * Sub-classes are encouraged to override this method.
+     *
+     * @param isLoaded
+     *            - a <code>boolean</code> that indicates a trailer is
+     *            currently loaded (if the value is <code>true</code>) or not
+     *            (if the value is <code>false</code>).
+     * @return a <code>String</code> describing the occupancy state of this
+     *         transporter.
+     */
+    protected String getVacancyString(boolean isLoaded) {
+        if (isLoaded) {
+            return TankTrailerHitch.NO_VACANCY_STRING;
+        }
+        return TankTrailerHitch.HAVE_VACANCY_STRING;
+    }
+
+    // Public constructors and methods.
+
+    /**
+     * Create a new hitch.
+     */
+    public TankTrailerHitch() {
+    }
+
+    /**
+     * Determines if this object can accept the given unit. The unit may not be
+     * of the appropriate type or there may be no room for the unit.
+     * <p>
+     *
+     * @param unit
+     *            - the <code>Entity</code> to be loaded.
+     * @return <code>true</code> if the unit can be loaded, <code>false</code>
+     *         otherwise.
+     */
+    public boolean canLoad(Entity unit) {
+        // Only trailers can be towed.
+        if (!(unit.isTrailer())) {
+            return false;
+        }
+
+        // We must have enough space for the trailer.
+        if (towed != Entity.NONE) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Load the given unit.
+     *
+     * @param unit
+     *            - the <code>Entity</code> to be loaded.
+     * @exception IllegalArgumentException
+     *                - If the unit can't be loaded, an
+     *                <code>IllegalArgumentException</code> exception will be
+     *                thrown.
+     */
+    public final void load(Entity unit) throws IllegalArgumentException {
+        // If we can't load the unit, throw an exception.
+        if (!canLoad(unit)) {
+            throw new IllegalArgumentException("Can not load " + unit.getShortName() + " onto this hitch.");
+        }
+
+        // Assign the unit as our carried troopers.
+        towed = unit.getId();
+    }
+
+    /**
+     * Get a <code>List</code> of the units currently loaded into this payload.
+     *
+     * @return A <code>List</code> of loaded <code>Entity</code> units. This
+     *         list will never be <code>null</code>, but it may be empty. The
+     *         returned <code>List</code> is independent from the under- lying
+     *         data structure; modifying one does not affect the other.
+     */
+    public final Vector<Entity> getLoadedUnits() {
```

Comment:
Make this `List<>`, we should not be introducing any new usages of `Vector<>`.

Reply:
Vector is the return type of the method this overrides, so a change to List would require changing it throughout the Transporter hierarchy.

Reply:
I do in fact try to get rid of Vectors when I find them. If I'm going to go through and update Transporter, I think that should be a separate PR for safety's sake.